File: java/pythagorean-triplet/src/main/java/PythagoreanTriplet.java
import java.util.List;

public class PythagoreanTriplet {
    private final int x;
    private final int y;
    private final int z;

    public PythagoreanTriplet(int x, int y, int z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

    public long calculateProduct() {
        return x * y * z;
    }

    public int calculateSum() {
        return x + y + z;
    }

    public boolean isPythagorean() {
        return x * x + y * y == z * z;
    }

    public static PythagoreanTripletBuilder makeTripletsList() {
        return new PythagoreanTripletBuilder();
    }
}

File: r/pythagorean-triplet/pythagorean-triplet.R
pythagorean_triplet <- function(n) {
  triplets = list()
  
  for (a in 1:(n/3-1)) {
    for (b in (a+1):((n-a)/2)) {
      c <- n - b - a
      if (a^2 + b^2 == c^2)
        triplets <- append(triplets, list(c(a, b, c)))
    }
  }
  
  triplets
}
